packages/core-ethereum/crates/core-ethereum-actions/src/channels.rs

```diff
@@ -44,6 +45,12 @@ where
         return Err(NotEnoughAllowance);
     }
 
+    if db.read().await.is_network_registry_enabled().await?
+        && !db.read().await.is_allowed_to_access_network(&destination).await?
+    {
+        return Err(PeerAccessDenied);
+    }
+
```

Comment:
This could be considered as a build feature flag and condition compile directly inside the DB operational callbacks.